core/experiments/assets/js/admin/behaviors/experiments-messages.js

```diff
@@ -109,45 +127,29 @@ export default class ExperimentsDependency {
 		return ! this.areAllDependenciesActive( dependencies );
 	}
 
-	showDependenciesDialog( experimentId ) {
-		const experiment = this.getExperimentData( experimentId ),
-			dependencies = this.getExperimentDependencies( experimentId );
+	isExperimentContainsDeactivatingMessage( experimentId ) {
+		return this.getExperimentDeactivatingDialogMessage( experimentId );
+	}
 
-		const dependenciesList = this.joinDepenednciesNames( dependencies.map( ( d ) => d.title ), ', ', ' & ' );
+	dialogActivateContentMessage( title, message ) {
+		// Translators: %1$s: Experiment title, %2$s: Message content
+		return __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
+			.replace( '%1$s', `<strong>${ title }</strong>` )
+			.replace( '%2$s', `<strong>${ message }</strong>` );
+	}
 
-		// Translators: %1$s: Experiment title, %2$s: Experiment dependencies list
-		const message = __( 'In order to use %1$s, first you need to activate %2$s.', 'elementor' )
-			.replace( '%1$s', `<strong>${ experiment.title }</strong>` )
-			.replace( '%2$s', `<strong>${ dependenciesList }</strong>` );
+	dialogDeactivateContentMessage( title, message ) {
+		// Translators: %1$s: Experiment title, %2$s: Message content
+		return __( 'While deactivating %1$s, %2$s.', 'elementor' )
+			.replace( '%1$s', `<strong>${ title }</strong>` )
+			.replace( '%2$s', `<strong>${ message }</strong>` );
+	}
```

Comment:
maybe remove

Reply:
Refactored, added the functionality to `getMessageDialogConfig()` and `getDependencyDialogConfig()`